### ISSUE TITLE
Revamp naming around `Poll`

### ISSUE THREAD

**Issue by aturon** [rust-lang-nursery/futures-rs#222]
We should consider renaming `Poll` as `AsyncResult`, since it ends up being used by methods not named `poll` (for example, `flush` in the upcoming `Sink` trait).

More generally, we should consider having a family of `Async*` types (we need more for `Sink`), such that returning something of type `Async*` means that you are "async aware", and hence that you schedule the current task for wakeup, etc.

**Comment by carllerche** [comment]
AsyncResult is pretty long though :(

**Comment by alexcrichton** [comment]
I'm slightly worried about getting too wordy here, `AsyncResult<Self::Item, Self::Error>` is a little large. That being said, I don't think this should block 0.2, we can rename at any time.

**Comment by aturon** [comment]
@alexcrichton To clarify, since this has come up on a couple of issues: I've been marking all breaking changes we've got under consideration as 0.2, not with the thought that they would "block" the release but that once we're ready to cut a breaking release, we should consider these various changes. Renamings etc get harder to do over time, so I think we should at least triage these before we cut a 0.2 release. Do you disagree? We could have a distinct label for proposed breakage, if you prefer.

**Comment by alexcrichton** [comment]
Yeah I'm not entirely certain what to do about tags and such. We have *so much* tagged with 0.2 (not just this repo, but others too) that it seems like a pipe dream to think that we'd get to everything. I'd personally prefer to stay focused and consider solely backwards incompatible changes with no deprecation story for 0.2. I think that issues that fall in the bucket of "not breaking, we can deprecate this" are fine to consider but shouldn't have the focus of our attention right now.

Now we haven't even really finished 0.1 by this point, though, so this is likely premature anyway.

**Comment by aturon** [comment]
Looks like we won't be taking this step in 0.2.
